### ISSUE TITLE
[BUG] Site menues turned to RTL and interface has issues

### ISSUE THREAD

**Issue by guymass** [WWBN/AVideo#8347]
I am not sure what happened but I assume this was a try to allow RTL interface on the site, moving the main menu to the right side and the top profile menu to the left. This is great however there are issue.

The main menu seem to float across the page from left to right to be opened on the right and when closed it floats back again to the left side of the screen. I cannot show it in an image screen shot, only in a video, maybe if you require it I will make one.

The second issue is the profile menu which is doing the opposite and is floating from the right side of the screen to the left side to be opened, but it is being over pushed out of the screen and is cut as in the following screen shot:

![Screenshot 5783-12-05 at 19 48 38](https://github.com/WWBN/AVideo/assets/58882903/0c0e32c7-6ce0-4743-a765-1ca06b5644bb)

This is a screenshot of how the main menu is displayed on the right side which seems good.
![Screenshot 5783-12-05 at 19 48 26](https://github.com/WWBN/AVideo/assets/58882903/b0416408-1381-4a56-b50e-1829a0eac59f)

however the content of the page is pushed under the menu and should be pushed more to the left.

**Comment by guymass** [comment]
Also the template seem to be stuck on this RTL as when I change my site the English language it will still open RTL. Maybe this is just a bug idk.

**Comment by DanielnetoDotCom** [comment]
I have no idea what the layout is that. Here is how it looks on my side

![image](https://github.com/WWBN/AVideo/assets/18009293/d7f79cfb-a7be-4854-8082-079316f24874)

**Comment by DanielnetoDotCom** [comment]
![chrome-capture-2023-7-23](https://github.com/WWBN/AVideo/assets/18009293/9b509bd6-cc3a-45a6-8447-048a01dda538)

**Comment by guymass** [comment]
Can you open my url and tell me if you get the same effect?

https://alive528.com

**Comment by guymass** [comment]
I see this youtube code added when I open the right side menu, it is floating in the wrong way and stuck on the template. I didn't add any custom code so idk:

![Screenshot 5783-12-07 at 8 45 41](https://github.com/WWBN/AVideo/assets/58882903/195c7d1a-a78d-4704-9087-89f5f5ce1fd9)
![Screenshot 5783-12-07 at 8 44 33](https://github.com/WWBN/AVideo/assets/58882903/41bb33d9-f8da-4eea-8620-4d753be2a749)

**Comment by DanielnetoDotCom** [comment]
yes all reversed and messed up on your URL

not sure, what you done. 

can you reproduce it on demo site?

**Comment by guymass** [comment]
Haven't done anything, only recent git pull updates and system upgrades, plugins etc... nothing out of the ordinary.

On my other site everything is normal. Can you tell me what could influence this behavior? This would take some css coding error to switch the direction of the menus. Where should I begin looking?
![Screenshot 5783-12-07 at 23 26 53](https://github.com/WWBN/AVideo/assets/58882903/5931ab4c-5237-477c-9eff-4df303c91e47)

**Comment by DanielnetoDotCom** [comment]
this is what I mean, I have no idea what is causing it. 

and you said you have no custom code. so I really have no idea. the only possibility is a custom CSS or JS code

**Comment by guymass** [comment]
can't find anything but this message in the console indicating something about the CSS direction, maybe this is our issue?
![Screenshot from 2023-08-26 01-51-38](https://github.com/WWBN/AVideo/assets/58882903/2346497d-a71a-475f-ada1-3aa6b8d25d0a)

**Comment by guymass** [comment]
What does this mean? maybe something in the way those values are calculated is wrong in the direction parameter?

This is from the browser's console:

When rendering the <html> element, the used values of CSS properties “writing-mode”, “direction”, and “text-orientation” on the <html> element are taken from the computed values of the <body> element, not from the <html> element’s own values. Consider setting these properties on the :root CSS pseudo-class. For more information see “The Principal Writing Mode” in https://www.w3.org/TR/css-writing-modes-3/#principal-flow

**Comment by guymass** [comment]
Also I see that the keyword "youtube" is added to the HTML tag when I open the menu, could this effect my interface?

html body.mainPage.isOnline.youtube

**Comment by guymass** [comment]
I even tried a hard reset and still nothing changes so I am not sure why is this still happening on my site.

**Comment by DanielnetoDotCom** [comment]
What do you mean by "hard reset"?

I must be honest, I have no idea what changes you have made, so I'm unable to help you revert it back. Additionally, I've been unable to reproduce the issue on my end.

Please note, if the AVideo script detects an RTL language, it automatically loads the following CSS: [https://github.com/WWBN/AVideo/blob/master/view/css/rtl.css](https://github.com/WWBN/AVideo/blob/master/view/css/rtl.css). If you've loaded any other CSS to handle RTL, that might be the root of the problem.

As a troubleshooting step, could you try accessing the site using a different browser and see if the problem persists?

**Comment by guymass** [comment]
I mean hard rest with the git pull command, meaning I loaded fresh copy of AVideo. I have nothing changed and nothing custom, im not loading any other RTL CSS file. This shows on all my browsers from safari to fireforx, chrome, brave, all of them. I really can't tell you more because this just happened one day after I updated the site.

**Comment by DanielnetoDotCom** [comment]
Firstly, I want to emphasize the importance of providing accurate information when seeking support. This aids in the swift resolution of the issue, and it helps in creating a smoother experience for both of us. Incorrect information can lead to unnecessary delays and confusion.

Upon inspecting your website, I found this specific CSS file: [https://alive528.com/atoms/wp-content/themes/enfold/css/rtl.css](https://alive528.com/atoms/wp-content/themes/enfold/css/rtl.css). This file is the likely cause of the RTL (right-to-left) issues you're experiencing. It appears to be custom and is not a part of the original AVideo script I've provided.

I had mentioned previously the possibility of a custom CSS affecting the functionality, and it seems that this is indeed the case. Please locate this file and assess its addition to your setup. 

I understand that you were certain there weren’t any custom modifications, but it's imperative for future interactions to thoroughly check and be certain of the information you provide. This will enable us to troubleshoot more effectively and avoid any potential misunderstandings.

**Comment by guymass** [comment]
Much thanks I will check, I didn't see this file at all and was sure we don't have any custom code there. Also im not sure why would this file effect the site as this never happened before. I will remove this and check again and hopefully this will resolve the issue and if so I  will close this.

**Comment by guymass** [comment]
Ok found it and fixed. It was still showing in my local cache, so once that has been deleted with coockies and refresh, its now back to they way it was. Much thanks, I wouldn't have noticed this at all. This was done about three years ago and then we removed it. Until recently we never had an issue with so we forgot it was there at all. In the past we tried to incorporate wordpress which was nice, but recent AVideo updates made us drop it and just use AVideo.

Again sorry for the confusion, I really forgot I have this code in there and it never bothered me before. I am closing this now.
